pkg/acs/calacs/acsccd/doccd.c

```diff
@@ -251,109 +257,291 @@ int DoCCD (ACSInfo *acs_info) {
         blevcorr[i] = PERFORM;
     }
 
-    /* default blev case, do original bias level subtraction */
-    if (acs_info->blevcorr == PERFORM &&
-        ((acs_info->expstart < SM4MJD && acs_info->detector == WFC_CCD_DETECTOR) ||
-         (acs_info->detector == WFC_CCD_DETECTOR && acs_info->subarray == YES) ||
-         (acs_info->detector != WFC_CCD_DETECTOR))) {
-        for (i = 0; i < acs_info->nimsets; i++) {
-            /* This is set based on results of FindOver */
-            done = overscan[i];
+    /* If this is subarray data, determine if it is supported for the bias shift 
+       correction.  Compare the aperture name against the array of supported
+       apertures.  Use the aperture name and not the date as the discriminant 
+       as the date is fuzzy - some supported subarrays were acquired before the 
+       official change, and some old style subarrays were acquired after the change.
 
-            if (doBlev(&acs[i], &x[i], acs[i].chip, &meanblev, &done,
-                       &driftcorr))
-                return (status);
+       ***FIX MDD: What about polarizer and ramp apertures?
+    */
+    Bool supportedSubAperture = NO;
```

Comment:
I feel that this should be inside the logic for WFC only, because this same code is also used to process HRC.

Reply:
Indeed!

Reply:
FYI: I've been trying to move stuff over to using ``stdbool.h::bool`` instead.